src/app/[locale]/(auth)/register/_hooks/use-register.ts

```
@@ -0,0 +1,24 @@
+export const useRegister = () => {
+    return useMutation({
+        mutationFn: async (data: RegisterInput) => {
+            try {
+                const res = await registerAction(data);
+
+                if (!res || res.error || res.error === "Failed") {
+                    throw new Error(res?.error || "Something went wrong");
+                }
+
+                return res;
+            } catch (err: any) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)